addons/mrp/models/mrp_production.py

```diff
@@ -670,10 +671,30 @@ def onchange_picking_type(self):
         self.location_src_id = self.picking_type_id.default_location_src_id.id or location.id
         self.location_dest_id = self.picking_type_id.default_location_dest_id.id or location.id
 
-    @api.onchange('qty_producing', 'lot_producing_id')
-    def _onchange_producing(self):
+    @api.onchange('qty_producing')
+    def _onchange_qty_producing(self):
         self._set_qty_producing()
 
+    @api.onchange('lot_producing_id')
+    def _onchange_lot_producing(self):
+        self._set_qty_producing()
+        if self.product_id.tracking == 'serial':
+            counter = self.env['stock.move.line'].search_count([
+                ('company_id', '=', self.company_id.id),
+                ('product_id', '=', self.product_id.id),
+                ('lot_id', '=', self.lot_producing_id.id),
+                ('move_id', 'in', self.env['stock.move'].search([
```

Comment:
To avoid another search() request,  I would make 
``` python
counter = self.env['stock.move.line'].search_count([
                ('company_id', '=', self.company_id.id),
                ('product_id', '=', self.product_id.id),
                ('lot_id', '=', self.lot_producing_id.id),
                ('state', '!=', 'cancel'),
                ('move_id', '!=', self.id)
])
```
The state, product and company are the same between stock.move and stock.move.line